src/components/approval-type/ApprovalType.vue

```diff
@@ -0,0 +1,169 @@
+import Null = types.Null;
+
+@Component({})
+export default class ApprovalType extends Vue {
+  // Refs
+  $refs!: Vue['$refs'] & {
+    courtNumRef: FormIF
+  }
+
+  ApprovalTypes = {
+    VIA_COURT_ORDER: 'VIA COURT ORDER',
+    VIA_REGISTRAR: 'VIA REGISTRAR'
+  }
+
+  /** Draft court order number. */
+  @Prop({ default: '' }) readonly draftCourtOrderNumber!: string
+
+  /** Whether approved by the registrar. */
+  @Prop({ default: false }) readonly approvedByRegistrar!: boolean
+
+  /** text used in radio options . */
+  @Prop({ default: false }) readonly isConversionToFullRestoration!: boolean
+
+  /** Show only the court order option; remove via registrar option. */
+  @Prop({ default: false }) readonly isCourtOrderOnly!: boolean
+
+  // Local properties
+  private courtOrderNumber = ''
+  private courtOrderNumRules = []
+  private valid = false
+  private approvalTypeSelected = ''
+
+  /** Called when component is mounted. */
+  mounted (): void {
+    // Copy props to mutable properties
+    if (this.approvedByRegistrar) {
+      this.courtOrderNumber = ''
+      this.approvalTypeSelected = this.ApprovalTypes.VIA_REGISTRAR
+    }
+    if (this.draftCourtOrderNumber) {
+      this.courtOrderNumber = this.draftCourtOrderNumber
+      this.approvalTypeSelected = this.ApprovalTypes.VIA_COURT_ORDER
+    }
+  }
+
+  /** Triggers the form validation. */
+  public validate (): boolean {
+    let status = this.$refs.courtNumRef.validate()
+    this.$emit('emitValid', status)
+    return status
+  }
+
+  private emitValidationError (event): void {
+    this.$emit('emitValid', !event)
+  }
+
+  @Emit('emitRadioButtonChange')
+  private radioButtonChanged (event): void {
+    if (event === this.ApprovalTypes.VIA_REGISTRAR) {
+      Vue.set(this, 'courtOrderNumber', '')
+      Vue.set(this, 'courtOrderNumRules', [])
+    }
+    if (event === this.ApprovalTypes.VIA_COURT_ORDER) {
+      this.setCourtNumberRules()
+    }
+    this.validate()
+  }
+
+  @Emit('emitCourtNumberChange')
+  private courtOrderNumberChanged (event): void {
+    Vue.set(this, 'approvalTypeSelected', this.ApprovalTypes.VIA_COURT_ORDER)
+    this.$emit('emitRadioButtonChange', this.ApprovalTypes.VIA_COURT_ORDER)
+    this.setCourtNumberRules()
+    this.validate()
+  }
+
+  private setCourtNumberRules (): void {
+    // Apply TextField rules
+    this.courtOrderNumRules = [
+      (v: string) => (!v || !/^\s/g.test(v)) || 'Invalid spaces', // leading spaces
+      (v: string) => (!v || !/\s$/g.test(v)) || 'Invalid spaces', // trailing spaces
+      (v: string) => (!v || !(v.length < 5)) || 'Court order number is invalid',
+      (v: string) => (!v || !(v.length > 20)) || 'Court order number is invalid',
+      (v: string) => !!v || 'A Court Order number is required'
+    ]
+  }
+
+  private getRadioText (option: string): string {
+    let restorationWording = 'restoration'
+    if (this.isConversionToFullRestoration) {
+      restorationWording = 'conversion to full restoration'
```

Comment:
If it's just a label changing, IMO you should pass the label string as a prop. It will keep this component more generic.

Reply:
Ah, good thought, I'll change it.

Reply:
Looks good.